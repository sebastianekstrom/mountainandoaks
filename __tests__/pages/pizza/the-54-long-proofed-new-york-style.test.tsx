import { render, screen } from "@testing-library/react";
import React from "react";

import The54 from "pages/pizza/the-54-long-proofed-new-york-style";

describe("The54 page", () => {
	it("renders title correctly", () => {
		render(<The54 />);
		const title = screen.queryAllByText(
			/The 54 â€“ A long proofed New York style pizza/i,
		);
		expect(title).toBeDefined();
	});

	it("renders the ingress text", () => {
		render(<The54 />);
		const ingress = screen.queryAllByText(
			/This recipe is like the cool cousin/i,
		);
		expect(ingress).toBeDefined();
	});

	it("renders the correct ingredients", () => {
		render(<The54 />);
		const ingredientList = screen.getByTestId("INGREDIENT_LIST");
		expect(ingredientList).toMatchSnapshot();
	});

	it("renders the method steps", () => {
		render(<The54 />);
		const step = screen.queryAllByText(
			/Warm up the water to room temperature/i,
		);
		expect(step).toBeDefined();
	});

	it("renders the alert with the correct description", () => {
		render(<The54 />);
		const alert = screen.queryAllByText(
			/Even though the container is coated in oil/i,
		);
		expect(alert).toBeDefined();
	});
});
