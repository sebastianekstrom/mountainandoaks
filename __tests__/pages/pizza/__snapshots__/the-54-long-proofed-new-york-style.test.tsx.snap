// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The54 page renders the correct ingredients 1`] = `
<section
  class="mb-8 lg:mb-0"
  data-testid="INGREDIENT_LIST"
>
  <div>
     
  </div>
  <div
    class="flex flex-col"
  >
    <ul>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Bread flour
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          164
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Water
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          103
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Salt
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          4.3
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Vegetable oil
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          3.4
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Dry yeast
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          0.1
          g
        </p>
      </li>
    </ul>
    <div
      class="ml-auto"
    >
      <button
        class="mt-2 rounded-full text-regular bg-pill-background py-2 px-4"
        data-testid="BAKERS_PERCENTAGE"
        role="button"
        type="button"
      >
        <p
          class="variable font-sans text-caption leading-paragraph "
        >
          Show bakers %
        </p>
      </button>
    </div>
  </div>
</section>
`;
